packages/aws-cdk/lib/api/bootstrap/bootstrap-template.yaml

```diff
@@ -233,6 +245,49 @@ Resources:
               - Ref: AWS::NoValue
       RoleName:
         Fn::Sub: cdk-${Qualifier}-image-publishing-role-${AWS::AccountId}-${AWS::Region}
+  LookupRole:
+    Type: AWS::IAM::Role
+    Properties:
+      AssumeRolePolicyDocument:
+        Statement:
+          - Action: sts:AssumeRole
+            Effect: Allow
+            Principal:
+              AWS:
+                Ref: AWS::AccountId
+          - Fn::If:
+              - HasTrustedAccountsForLookup
```

Comment:
We need 2 of these blocks: one for regular "trusted" accounts and one for accounts trusted for lookups.

Reply:
Aha, I see you did it in the CLI instead. 

For my money, I think it makes more sense to put this logic in the template. It makes for a more sensible interface imo, and if we put the logic in the template, then people who take the template and deploy it via StackSets will also be able to benefit from the better UX.